src/Hungary/Mews.Fiscalizations.Hungary.Tests/InvoiceTests.cs

```diff
@@ -1,184 +1,336 @@
+            var sendInvoiceResponse = await navClient.SendInvoicesAsync(
+                token: exchangeToken.SuccessResult,
+                invoices: Sequence.FromPreordered(new[] { invoice }, startIndex: 1).Get()
+            );
+
+            Thread.Sleep(2000);
+
+            var sendInvoiceTransactionStatus = await navClient.GetTransactionStatusAsync(sendInvoiceResponse.SuccessResult);
+            TestFixture.AssertResponse(sendInvoiceTransactionStatus);
+            AssertInvoiceStatuses(sendInvoiceTransactionStatus.SuccessResult.InvoiceStatuses);
+
+            var sendModificationInvoiceResponse = await navClient.SendModificationDocumentsAsync(
+                token: exchangeToken.SuccessResult,
+                invoices: Sequence.FromPreordered(new[] { CreateModificationInvoice(invoice.Number, receiver) }, startIndex: 1).Get()
+            );
+
+            Thread.Sleep(2000);
+
+            var sendModificationInvoiceTransactionStatus = await navClient.GetTransactionStatusAsync(sendModificationInvoiceResponse.SuccessResult);
+            TestFixture.AssertResponse(sendModificationInvoiceTransactionStatus);
+            AssertInvoiceStatuses(sendModificationInvoiceTransactionStatus.SuccessResult.InvoiceStatuses);
+        }
+
+        [Test, Order(1)]
+        [Ignore("Will be rewritten and re-enabled when upgrading to V3.0.")]
+        [TestCase("CZ", "CZ12345678")]
+        [TestCase("US", "UsTaxId")]
+        [TestCase("CZ", null)]
+        [TestCase("US", null)]
+        public async Task SendCorrectionForeignCompanyInvoiceSucceeds(string countryCode, string taxId)
         {
             var navClient = TestFixture.GetNavClient();
             var exchangeToken = await navClient.GetExchangeTokenAsync();
-            var response = await navClient.SendModificationDocumentsAsync(
+            var country = Countries.GetByCode(countryCode).Get();
+            var taxpayerNumber = taxId.ToNonEmptyOption().Map(i => TaxpayerIdentificationNumber.Create(country, i).Success.Get());
+            var foreignCompany = ForeignCompany.Create(
+                name: Name.Create("Foreign test company ltd.").Success.Get(),
+                address: CreateAddress(country),
+                taxpayerId: taxpayerNumber.GetOrNull()
+            );
+            var receiver = new Receiver(new Company(foreignCompany.Success.Get()));
+            var invoice = CreateInvoice(receiver);
+            var sendInvoiceResponse = await navClient.SendInvoicesAsync(
+                token: exchangeToken.SuccessResult,
+                invoices: Sequence.FromPreordered(new[] { invoice }, startIndex: 1).Get()
+            );
+
+            Thread.Sleep(2000);
+
+            var sendInvoiceTransactionStatus = await navClient.GetTransactionStatusAsync(sendInvoiceResponse.SuccessResult);
+            TestFixture.AssertResponse(sendInvoiceTransactionStatus);
+            AssertInvoiceStatuses(sendInvoiceTransactionStatus.SuccessResult.InvoiceStatuses);
+
```

Comment:
There's quite a lot of code duplication. Could we just have a parametrized helper for this?

Reply:
As discussed, I am planning to refactor the tests later when the NAV test UI is back online.